internal/bminventory/inventory.go

```diff
@@ -768,23 +768,30 @@ func (b *bareMetalInventory) getNewClusterCPUArchitecture(newClusterParams *mode
 }
 
 func (b *bareMetalInventory) RegisterAddHostsCluster(ctx context.Context, params installer.RegisterAddHostsClusterParams) middleware.Responder {
-	c, err := b.RegisterAddHostsClusterInternal(ctx, nil, params, true)
+	v2Params := installer.V2ImportClusterParams{
+		NewImportClusterParams: &models.ImportClusterParams{
+			APIVipDnsname:    params.NewAddHostsClusterParams.APIVipDnsname,
+			Name:             params.NewAddHostsClusterParams.Name,
+			OpenshiftVersion: params.NewAddHostsClusterParams.OpenshiftVersion,
+		},
+	}
+	c, err := b.V2ImportClusterInternal(ctx, nil, params.NewAddHostsClusterParams.ID, v2Params, true)
 	if err != nil {
 		return common.GenerateErrorResponder(err)
 	}
 	return installer.NewRegisterAddHostsClusterCreated().WithPayload(&c.Cluster)
 
 }
-func (b *bareMetalInventory) RegisterAddHostsClusterInternal(ctx context.Context, kubeKey *types.NamespacedName, params installer.RegisterAddHostsClusterParams, v1Flag bool) (*common.Cluster, error) {
-	id := params.NewAddHostsClusterParams.ID
+func (b *bareMetalInventory) V2ImportClusterInternal(ctx context.Context, kubeKey *types.NamespacedName, id *strfmt.UUID,
```

Comment:
perhaps rename `id` to something like `OCPClusterId`?
this is a bit confusing since we have basically the AI cluster ID (which we generate) and the OCP cluster ID (which the user should provide). so hard to know which one is it

Reply:
Fixed by adding a new input parameter OpenshiftClusterID

Reply:
great, thanks.